### PR TITLE
added "--add-opens" option to some renaissance benchmarks

### PR DESCRIPTION
**Pull request by norbline**
Referred to #2500  
--add-opens were required to make the benchmarks run on JDK16+

### CODE REVIEW

**Comment by norbline** [comment]
link to grinder https://ci.adoptopenjdk.net/job/Grinder/2991/

**Review by smlambert — COMMENTED** [review]
We will need to deal with jdk8 version differently as --add-opens options are only valid for jdk11+ (see https://ci.adoptopenjdk.net/job/Grinder/3000/tapResults/ which is a jdk8 run).  

One way this has been done for other tests is to update the ADD_OPENS_CMD in a perf.mk make file, similar to what was done for system tests (see https://github.com/adoptium/aqa-tests/blob/master/system/systemtest.mk#L60-L63 and https://github.com/adoptium/aqa-tests/blob/master/perf/perf.mk#L3, in this way, the options will be present for jdk11+ and absent for jdk8).

**Comment by norbline** [comment]
Oh okay 😊..thanks

File: perf/renaissance/playlist.xml [hunk 1]
```diff
@@ -94,7 +94,7 @@
 				<platform>x86-32_windows|arm_linux</platform>
 			</disable>
 		</disables>
-		<command>$(JAVA_COMMAND) --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.util.concurrent=ALL-UNNAMED --add-opens java.base/java.nio=ALL-UNNAMED --add-opens java.base/sun.nio.ch=ALL-UNNAMED  -jar $(Q)$(TEST_RESROOT)$(D)renaissance.jar$(Q) --json $(Q)$(REPORTDIR)$(D)dec-tree.json$(Q) dec-tree; \
+		<command>$(JAVA_COMMAND) -jar $(Q)$(TEST_RESROOT)$(D)renaissance.jar$(Q) --json $(Q)$(REPORTDIR)$(D)dec-tree.json$(Q) dec-tree; \
```
Comment (sophia-guo) [inline perf/renaissance/playlist.xml:97]: `$(ADD_OPENS_CMD)` is missing?

Comment (norbline) [reply perf/renaissance/playlist.xml:97]: yea wow okay is like i omitted that

Comment (norbline) [reply perf/renaissance/playlist.xml:97]: no changes ... same error as before. 
 grinder: https://ci.adoptopenjdk.net/job/Grinder/3035/console

**Comment by sophia-guo** [comment]
The grinder is against `renaissance-chi-square`. The command is set with `ADD_OPENS_CMD` 

```
		<testCaseName>renaissance-chi-square</testCaseName>
		<disables>
			<disable>
				<comment>https://github.com/adoptium/aqa-tests/issues/3189</comment>
				<platform>x86-32_windows|arm_linux</platform>
			</disable>
		</disables>
		<command>$(JAVA_COMMAND) $(ADD_OPENS_CMD) -jar $(Q)$(TEST_RESROOT)$(D)renaissance.jar$(Q) --json $(Q)$(REPORTDIR)$(D)chi-square.json$(Q) chi-square; \
		$(TEST_STATUS)</command>
```

However testoutput says the command is 

```
16:59:05  	"/Users/jenkins/workspace/Grinder/openjdkbinary/j2sdk-image/Contents/Home/bin/..//bin/java"  -jar "/Users/jenkins/workspace/Grinder/aqa-tests/TKG/../../jvmtest/perf/renaissance/renaissance.jar" --json ""/Users/jenkins/workspace/Grinder/aqa-tests/TKG/../TKG/output_16425431455575/renaissance-chi-square_0"/chi-square.json" chi-square; \
```

That is `$(ADD_OPENS_CMD)` wasn't parsed at all. Need to check why?

**Comment by sophia-guo** [comment]
Checked regular run looks like `$(ADD_OPENS_CMD)` was parsed correctly either.

 https://ci.adoptopenjdk.net/view/Test_perf/job/Test_openjdk17_hs_extended.perf_x86-64_mac/41/

```
14:41:50  ===============================================
14:41:50  Running test renaissance-finagle-http_0 ...
14:41:50  ===============================================
14:41:50  renaissance-finagle-http_0 Start Time: Sun Jan 16 19:41:49 2022 Epoch Time (ms): 1642362109902
14:41:50  Nothing to be done for setup.
14:41:50  variation: NoOptions
14:41:50  JVM_OPTIONS:  
14:41:50  { itercnt=1; \
14:41:50  	mkdir -p "/Users/jenkins/workspace/Test_openjdk17_hs_extended.perf_x86-64_mac/aqa-tests/TKG/../TKG/output_16423620768701/renaissance-finagle-http_0"; \
14:41:50  	cd "/Users/jenkins/workspace/Test_openjdk17_hs_extended.perf_x86-64_mac/aqa-tests/TKG/../TKG/output_16423620768701/renaissance-finagle-http_0"; \
14:41:50  	"/Users/jenkins/workspace/Test_openjdk17_hs_extended.perf_x86-64_mac/openjdkbinary/j2sdk-image/Contents/Home/bin/..//bin/java"  -jar "/Users/jenkins/workspace/Test_openjdk17_hs_extended.perf_x86-64_mac/aqa-tests/TKG/../../jvmtest/perf/renaissance/renaissance.jar" --json ""/Users/jenkins/workspace/Test_openjdk17_hs_extended.perf_x86-64_mac/aqa-tests/TKG/../TKG/output_16423620768701/renaissance-finagle-http_0"/finagle-http.json" finagle-http; \

```

**Comment by norbline** [comment]
Is like I am missing something which I can't figure out

**Comment by sophia-guo** [comment]
Can you try:

```
ADD_OPENS_CMD:=
ifneq ($(JDK_VERSION),8)
	ADD_OPENS_CMD=--add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.util.concurrent=ALL-UNNAMED --add-opens java.base/java.nio=ALL-UNNAMED --add-opens java.base/sun.nio.ch=ALL-UNNAMED --add-opens java.base/java.lang.invoke=ALL-UNNAMED
endif
```

The main difference is remove $(Q)

**Comment by sophia-guo** [comment]
`Is like I am missing something which I can't figure out`

No, it's not your change. The issue was there before. $(ADD_OPENS_CMD) was incorrectly parsed either in regular run.

**Comment by norbline** [comment]
i will try that, i will test locally first before committing changes
 you mean ADD_OPENS_CMD:=

**Comment by sophia-guo** [comment]
ADD_OPENS_CMD:=

**Comment by norbline** [comment]
output command after removing $(Q)
`/home/norbline/jdk-17.0.1+12/bin/java"  -jar "/home/norbline/aqa-testsA/TKG/../../jvmtest/perf/renaissance/renaissance.jar" --json ""/home/norbline/aqa-testsA/TKG/../TKG/output_16426043927767/renaissance-als_0"/als.json" als; \
`

**Comment by sophia-guo** [comment]
Yeah, we need to  add `<include>../perf.mk</include>` before line 16 https://github.com/adoptium/aqa-tests/blob/master/perf/renaissance/playlist.xml#L16

**Comment by norbline** [comment]
All tests passed
-renaissance-dec-tree
Grinder link: https://ci.adoptopenjdk.net/job/Grinder/3047/
-renaissance-als
 Grinder link: https://ci.adoptopenjdk.net/job/Grinder/3057/
-renaissance-movie-lens
 Grinder link: https://ci.adoptopenjdk.net/job/Grinder/3054/
-renaissance-chi-square
 Grinder linK: https://ci.adoptopenjdk.net/job/Grinder/3060/
-renaissance-log-regression
 Grinder link: https://ci.adoptopenjdk.net/job/Grinder/3061/
-renaissance-gauss-mix
 Grinder link: https://ci.adoptopenjdk.net/job/Grinder/3058/

**Review by sophia-guo — APPROVED** [review]

**Review by smlambert — APPROVED** [review]
thanks @norbline!

**Pull request merged**